src/solver/src/steady_mesh_adaptive_solver.C

```diff
@@ -110,6 +112,28 @@ namespace GRINS
             error.plot_error( this->_error_plot_prefix+".exo", mesh );
           }
 
+	// Get the global error estimate if you can and are asked to
+	if( this->compute_QoI_error_estimate )
+	{
+	  // First check if we are using the Adjoint Refinement Error Estimator
+	  if(context.error_estimator->type() == libMesh::ADJOINT_REFINEMENT)
+	  {
+	    // Loop over QoIs and print error error estimates
+	    for(unsigned int i = 0; i != context.system->qoi.size(); i++)
+	    {
+	      libMesh::AdjointRefinementEstimator* adjoint_ref_error_estimator = libMesh::libmesh_cast_ptr<libMesh::AdjointRefinementEstimator*>( context.error_estimator.get() );
+	      std::cout<<"The error estimate for QoI("<<i<<") is: "<<adjoint_ref_error_estimator->get_global_QoI_error_estimate(i)<<std::endl;
+	    }
+
+	  }
+	  else
+	  {
+	    std::string warning = "You asked for QoI error estimates but did not use an Adjoint Refinement Error Estimator!\n";
```

Comment:
Why this warning? Seems perfectly reasonable to me to want to use the AdjointResidual estimator to drive some adaptive refinement.